ui/ui-components/pages/model/[modelId]/source/[sourceId]/edit.tsx

```diff
@@ -289,20 +338,13 @@ const Page: NextPage<Props> = ({
   };
 
   const updateOption = async (optKey, optValue) => {
-    const _source = Object.assign({}, source);
+    const _source = { ...source };
     _source.options[optKey] = optValue;
+    _source.mapping = {};
```

Comment:
why is this necessary?

Reply:
Keeping it means when you change your table source it _sticks_ to whatever mapping value you have. Clearing this effectively resets that.